service/policy/keymanagement/key_management.go

```diff
@@ -0,0 +1,173 @@
+	policyconfig "github.com/opentdf/platform/service/policy/config"
+	policydb "github.com/opentdf/platform/service/policy/db"
+)
+
+type Service struct {
+	dbClient policydb.PolicyDBClient
+	logger   *logger.Logger
+	config   *policyconfig.Config
+}
+
+func NewRegistration(ns string, dbRegister serviceregistry.DBRegister) *serviceregistry.Service[keyMgmtConnect.KeyManagementServiceHandler] {
+	return &serviceregistry.Service[keyMgmtConnect.KeyManagementServiceHandler]{
+		ServiceOptions: serviceregistry.ServiceOptions[keyMgmtConnect.KeyManagementServiceHandler]{
+			Namespace:      ns,
+			DB:             dbRegister,
+			ServiceDesc:    &keyMgmtProto.KeyManagementService_ServiceDesc,
+			ConnectRPCFunc: keyMgmtConnect.NewKeyManagementServiceHandler,
+			RegisterFunc: func(srp serviceregistry.RegistrationParams) (keyMgmtConnect.KeyManagementServiceHandler, serviceregistry.HandlerServer) {
+				cfg, err := policyconfig.GetSharedPolicyConfig(srp.Config)
+				if err != nil {
+					srp.Logger.Error("Failed to get shared policy config", slog.String("error", err.Error()))
+					panic(err)
+				}
+				ksvc := &Service{
+					dbClient: policydb.NewClient(srp.DBClient, srp.Logger, int32(cfg.ListRequestLimitMax), int32(cfg.ListRequestLimitDefault)),
+					logger:   srp.Logger,
+					config:   cfg,
+				}
+				return ksvc, nil
+			},
+		},
+	}
+}
+
+func (ksvc Service) CreateProviderConfig(ctx context.Context, req *connect.Request[keyMgmtProto.CreateProviderConfigRequest]) (*connect.Response[keyMgmtProto.CreateProviderConfigResponse], error) {
+	rsp := &keyMgmtProto.CreateProviderConfigResponse{}
+
+	ksvc.logger.Debug("Creating Provider Config")
+
+	auditParams := audit.PolicyEventParams{
+		ActionType: audit.ActionTypeCreate,
+		ObjectType: audit.ObjectTypeKeyManagementProviderConfig,
+	}
+
+	pc, err := ksvc.dbClient.CreateProviderConfig(ctx, req.Msg)
+	if err != nil {
+		ksvc.logger.Audit.PolicyCRUDFailure(ctx, auditParams)
+		return nil, db.StatusifyError(err, db.ErrTextCreationFailed, slog.String("keyManagementService", req.Msg.GetName()))
+	}
+
+	auditParams.ObjectID = pc.GetId()
+	auditParams.Original = pc
+	ksvc.logger.Audit.PolicyCRUDSuccess(ctx, auditParams)
+
+	rsp.ProviderConfig = pc
+
+	return connect.NewResponse(rsp), nil
+}
+
+func (ksvc Service) GetProviderConfig(ctx context.Context, req *connect.Request[keyMgmtProto.GetProviderConfigRequest]) (*connect.Response[keyMgmtProto.GetProviderConfigResponse], error) {
+	rsp := &keyMgmtProto.GetProviderConfigResponse{}
+
+	switch req := req.Msg.GetIdentifier().(type) {
+	case *keyMgmtProto.GetProviderConfigRequest_Id:
+		ksvc.logger.Debug("Getting Provider config by ID", slog.String("ID", req.Id))
+	case *keyMgmtProto.GetProviderConfigRequest_Name:
+		ksvc.logger.Debug("Getting Provider config by Name", slog.String("Name", req.Name))
+	default:
+		return nil, connect.NewError(connect.CodeInvalidArgument, nil)
+	}
+
+	auditParams := audit.PolicyEventParams{
```

Comment:
You may disagree, but I don't think we need audit logs for `read` events here, as we don't utilize them throughout the rest of the policy services.